pilot/pkg/model/sidecar.go

```diff
@@ -663,6 +664,36 @@ func serviceMatchingListenerPort(service *Service, ilw *IstioEgressListenerWrapp
 	return nil
 }
 
+func serviceMatchingVirtualServicePorts(service *Service, vsDestPorts sets.IntSet) *Service {
+	// A value of 0 in vsDestPorts is used as a sentinel to indicate a dependency
+	// on every port of the service.
+	if len(service.Ports) == 1 || len(vsDestPorts) == 0 || vsDestPorts.Contains(0) {
```

Comment:
I don't think it matters much in practice, but is `len(service.Ports) == 1` actually correct?

Imagine I have a Svc with port=1234 and a VS pointing to svc:5678. I think this would select the service?

Reply:
Yes, this would select the service in that case. You're right, I think this is probably incorrect, and should be fine to remove. Will update